### ISSUE TITLE
[Enhancement] IO Buffer

### ISSUE THREAD

**Issue by NickFillot** [h2oai/datatable#3093]
Hello, is it possible to implement natively io write ?

For the moment i use this workaround, but maybe there is a better way ?

```python
import pandas, typing

def to_jay(df: pandas.DataFrame, object_: object,  *args: typing.Any, **kwargs: typing.Any) -> None:
        if hasattr(object_, 'write'):
            with tempfile.TemporaryDirectory() as tmpd:
                buf = os.path.join(tmpd, 'df.jay')
                Frame(df).to_jay(buf, *args, **kwargs)
                with open(buf, 'rb') as tmpf:
                    object_.write(tmpf.read())
        else:
            Frame(df).to_jay(*args, **kwargs)
```

And maybe also implement storage_options parameter pandas-like ?

Br
Nick

**Comment by st-pasha** [comment]
So, if I understand your question correctly, you want the following to work:
```
df.to_jay(writeable)
```
where `df` is a datatable Frame, and `writeable` is any python object that supports `.write(bytes)` interface.

Or, with `storage_options`:
```
df.to_jay(writeable, storage_options=options)
```
where `writeable` object implements interface `.write(bytes, **options)`.

Is my understanding correct here? What are the examples when storage options would be useful?

**Comment by NickFillot** [comment]
Yes, it would be indeed write in writable objects (as you also read from readable objects like files)

storage_options would be a bonus on my side, it would be to communicate with remote files hosted remote file systems (s3, hdfs, etc), pandas uses [fsspec](https://filesystem-spec.readthedocs.io/en/latest/) to write with parameters simply stored in a dict

But writable object should offer enough customizable development (SO: S3 read)[https://stackoverflow.com/questions/37703634/how-to-import-a-text-file-on-aws-s3-into-pandas-without-writing-to-disk]
